### ISSUE TITLE
wrong namespace used for the protein modification ontology

### ISSUE THREAD

**Issue by sbmlsecretary** [sbmlteam/sbml-specifications#291]
Apply to SBML Level 2 Version 5 release 1 and SBML Level 3 Version 2 draft (August 2014)

In page 105 of the L2V5 specs, there is an error in one of the example of nested annotations. Identifiers.org url should be used and more importantly the namespace used for the protein modification ontology is wrong (it is not 'obo.mod' but 'psimod').

Here is the xml in the specs:

<species id="S1" metaid="_000004" compartment="cell"
hasOnlySubstanceUnits="false" boundaryCondition="false"
constant="false">
<annotation>
<rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
xmlns:bqbiol="http://biomodels.net/biology-qualifiers/">
<rdf:Description rdf:about="#_000004" >
<bqbiol:isVersionOf>
<rdf:Bag>
<rdf:li rdf:resource="urn:miriam:uniprot:P04551"/>
<bqbiol:isVersionOf>
<rdf:Bag>
<rdf:li rdf:resource="urn:miriam:obo.mod:MOD%3A00047"/>
</rdf:Bag>
</bqbiol:isVersionOf>
</rdf:Bag>
</bqbiol:isVersionOf>
</rdf:Description>
</rdf:RDF>
</annotation>
</species>

It should be changed to:

<species id="S1" metaid="_000004" compartment="cell"
hasOnlySubstanceUnits="false" boundaryCondition="false"
constant="false">
<annotation>
<rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
xmlns:bqbiol="http://biomodels.net/biology-qualifiers/">
<rdf:Description rdf:about="#_000004" >
<bqbiol:isVersionOf>
<rdf:Bag>
<rdf:li rdf:resource="http://identifiers.org/uniprot/P04551"/>
<bqbiol:isVersionOf>
<rdf:Bag>
<rdf:li rdf:resource="http://identifiers.org/psimod/MOD:00047"/>
</rdf:Bag>
</bqbiol:isVersionOf>
</rdf:Bag>
</bqbiol:isVersionOf>
</rdf:Description>
</rdf:RDF>
</annotation>
</species>

The same problem happen in the L3V2 draft, on page 108-109

Thanks

Reported by: niko-rodrigue

Original Ticket: [sbml/sbml-specifications//293](https://sourceforge.net/p/sbml/sbml-specifications//293)

**Comment by sbmlsecretary** [comment]
- **labels**:  --> Level 3 Version 1 Core
- **status**: open --> pending
- **assigned_to**: Lucian Smith
- **Group**: Reported-Proposed --> Accept-formatting-and-typos


Original comment by: luciansmith

**Comment by sbmlsecretary** [comment]
Thanks!  This is now fixed in SVN for L3v2, and will be incorporated into the forthcoming specification.

I'm not sure what the best thing to do for L2v5 is.  Would it be possible to make this change, and incorporate it into a new *release* of the spec, instead of a whole new version?

Original comment by: luciansmith

**Comment by sbmlsecretary** [comment]
Yes, I believe it would be acceptable to issue a new release of the specification. The SBML Development Process says this about releases, and making a new release for this issue is perfectly in line with this principle:

"After a period of time, an accumulation of issues may warrant the development of a new release of Version 4, ..."

(and yes, it needs to be updated to talk about version 5.)

Original comment by: mhucka

**Comment by sbmlsecretary** [comment]
- **labels**: Level 3 Version 1 Core --> Level 2 Version 5 Core
- **status**: pending --> accepted


Original comment by: luciansmith

**Comment by sbmlsecretary** [comment]
With the release of L3v2, this issue is fixed there.  However, it remains in l2v5, so I added a tag for that specification instead, so we can add it to the list of errata there, possibly for a later release of that spec.

Original comment by: luciansmith

**Comment by sbmlsecretary** [comment]
This is now fixed in the SVN for L2v5 as well; whenever we do a new release of that, those changes will go in.

Original comment by: luciansmith

**Comment by sbmlsecretary** [comment]
Please update the identifiers urls to the https versions. Unsecure connections should be avoided whenever possible (and could also become unavailable at some point, e.g., as happened when EBI moved completely to https).

Otherwise agree with the change.

Original comment by: matthiaskoenig

**Comment by sbmlsecretary** [comment]
I agree with Matthias - change to https and then okay

Original comment by: sarahkeating

**Comment by sbmlsecretary** [comment]
I agree with this change

Original comment by: chaouiya

**Comment by sbmlsecretary** [comment]
There are a lot of 'http://' lines in the SBML spec, and I'm not sure how many can and/or should be changed to 'https://'.   Can we get away with changing all of them except the namespace URLs?

Original comment by: luciansmith

**Comment by sbmlsecretary** [comment]
I agree with the change. Regarding http://identifiers.org or https://identifiers.org in namespace urls, I would not consider it is a major issue at least for now. Both ways would be ok. The identifiers.org website now redirects http to https and this redirection service should be a feature maintained by identifier.org for a long term.

Original comment by: zhangfen

**Comment by sbmlsecretary** [comment]
OK, went through all the URLs in the spec and updated them to use https if possible.  This was, notably, *not* possible with sbml.org and biomodels.net and a few others; identifiers.org was one of the few ones where https actually worked.  However, that was for l3 and l2; adding a new issue for that.

Original comment by: luciansmith

**Comment by sbmlsecretary** [comment]
Hi Lucian, 
I just had a look at the spec. Yes, namespaces must stay, but many links can be updated. Here my impression:

**Replace**
http://www.biopax.org/ -> https://www.biopax.org/ (url, https exists)
http://www.uniprot.org/* -> https://www.uniprot.org/*
http://identifiers.org/* -> https://identifiers.org/*
http://www.ebi.ac.uk/* -> https://www.ebi.ac.uk/*
http://www.genome.jp/dbget-bin/www_bget?ec:3.5.4.4 -> https://www.genome.jp/dbget-bin/www_bget?ec:3.5.4.4
http://www.cellml.org/specification -> https://www.cellml.org/specification
http://tools.ietf.org/html/rfc6350 -> https://tools.ietf.org/html/rfc6350

**Remove**
http://www.godatabase.org/cgi-bin/amigo/go.cgi?view=details&query=GO:0007268 (link is dead, resource probably does not exist any more)
http://us.expasy.org/cgi-bin/nicezyme.pl?3.5.4.4 (link is dead, resource does not exist any more)
http://www.math.uwaterloo.ca/bingalls/Pubs/conservation.pdf (link is dead, resource does not exist anymore)
.
**Not sure about**
http://biomodels.net/sbo/ -> https://www.ebi.ac.uk/sbo/main/ (forwards, https exists, careful this is also part of namespaces, only replace when the website is mentioned)

**Don't replace**
http://sbml.org/* (urls, no https yet)
http://www.sbml.org/sbml/* (namespace)
http://www.w3.org/* (namespace)
http://www.mysim.org/ns (namespace)
http://www.struct.org/ns (namespace)
http://www.othersim.com/ (namespace)
http://www.sbgn.org/ (url, no https yet)
http://www.psidev.info/index.php?q=node/60 (url, no https yet)
http://biomodels.net/* (used in definitionURL)
http://co.mbine.org/* (no https yet, part of namespaces)
http://purl.org/dc/* (namespace)
http://www.informatics.jax.org/searches/GO.cgi?id=GO:0007268 (url, no https yet)
http://dublincore.org/* (url, no https yet)

Hope this helps.
M

.






Original comment by: matthiaskoenig

**Comment by sbmlsecretary** [comment]
- **status**: accepted --> pending


Original comment by: luciansmith

**Comment by sbmlsecretary** [comment]
Accepted and pending for l2v5.  Also added to http://sbml.org/Documents/Specifications/SBML_Level_2/Version_5/Collected_Errata_to_the_SBML_Level_2_Version_5_Specification

Original comment by: luciansmith

**Comment by luciansmith** [comment]
As these changes have been made to the source code, following our new procedures, I'm going ahead and closing the issue here, as we no longer have a 'pending' status between the issue being addressed and the spec being released.
